### PR TITLE
Add controller support

### PR DESCRIPTION
**Pull request by nicoeps**
Add controller support because why not

### CODE REVIEW

**Comment by krs013** [comment]
Like I said on discord, this is an awesome feature that I've wanted for a while, and it works great on my machine so far. Haven't tested on anything besides linux but I can later. We may want to discuss a better place for `_sdlcontroller` than as a global variable but otherwise I think this is a pretty easy one to accept!

(Also I haven't looked for myself yet but will this work with `pyboy.PyBoy.get_input()`?)

File: pyboy/plugins/window_sdl2.pxd [hunk 1]
```diff
@@ -12,9 +12,10 @@ cimport cython
 from libc.stdint cimport uint8_t, uint16_t, int16_t, uint32_t
 
 
-
 cdef int ROWS, COLS
 
+cdef sdl2.SDL_GameController *_sdlcontroller = NULL
```
Comment (Baekalfen) [inline pyboy/plugins/window_sdl2.pxd:17]: Could we move this into the `WindowSDL2` class? Then I'm ready for merging this

Comment (Baekalfen) [reply pyboy/plugins/window_sdl2.pxd:17]: Ok, I see the problem. We'll just merge it as is

**Comment by krs013** [comment]
Before we merge I'd like to quickly test how it works on macOS and Windows. Also, are we happy with the button mapping? I don't have any objections, just mentioning.

**Comment by Baekalfen** [comment]
> Before we merge I'd like to quickly test how it works on macOS and Windows. Also, are we happy with the button mapping? I don't have any objections, just mentioning.

You're welcome to try it out, I'll wait a bit. I've looked at the mapping again. I don't have much of an opinion, but it looks reasonable?

**Comment by krs013** [comment]
I can't get the controller to work on macOS. There's a lot that could be going wrong but I have confirmed that the controller works in at least one other game. Has anyone else had success with macOS?

Linux works perfectly, and Windows works, but it swaps the B/A keys because I have a Wii U controller on an Xinput profile. We might want format switching in the future but I'm fine leaving that out of the scope of this PR. Also, using the home button for speed up activated Steam Big Picture and crashed PyBoy, but after I quit Steam it was fine. So I'm not sure if we need to change that or just punt it to configuration too.

If anyone can confirm it works on macOS then I'm fine to merge.

**Comment by Baekalfen** [comment]
My wireless Xbox 360 controller doesn't work on Mac, so I can't test it. But I'm willing to merge it as-is, as any issues on Mac, likely would be because of SDL, and not PyBoy.

**Pull request merged**